include/pybind11/eigen.h

```diff
@@ -641,6 +643,343 @@ struct type_caster<Type, enable_if_t<is_eigen_other<Type>::value>> {
     using cast_op_type = Type;
 };
 
+template <int Options>
+struct eigen_to_numpy {};
+
+template <>
+struct eigen_to_numpy<Eigen::ColMajor> {
+    static constexpr int flag = array::f_style;
+};
+
+template <>
+struct eigen_to_numpy<Eigen::RowMajor> {
```

Comment:
Do these actually need to be helper structs? It feels like this can be done with constexpr logic entirely even in C++11. 

Also there is no static_assert failures here to guard against accidental SFINAE. Not to mention the naming here is a bit generic / problematic as it doesn't imply that these are used specifical for Eigen Tensor conversions.